node/_fs/_fs_truncate_test.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+import { assertEquals, assertThrows, fail } from "../../testing/asserts.ts";
+import { truncate, truncateSync } from "./_fs_truncate.ts";
+
+Deno.test({
+  name: "ASYNC: no callback function results in Error",
+  fn() {
+    assertThrows(
+      () => {
+        truncate("some/path", 0);
+      },
+      Error,
+      "No callback function supplied",
+    );
+  },
+});
+
+Deno.test({
+  name: "ASYNC: truncate entire file contents",
+  async fn() {
+    const file: string = Deno.makeTempFileSync();
+    await Deno.writeFile(file, new TextEncoder().encode("hello world"));
+
+    await new Promise<void>((resolve, reject) => {
+      truncate(file, (err: Error | null) => {
+        if (err !== null) reject();
+        else resolve();
+      });
+    })
+      .then(
+        () => {
+          const fileInfo: Deno.FileInfo = Deno.lstatSync(file);
+          assertEquals(fileInfo.size, 0);
+        },
+        () => {
+          fail("No error expected");
+        },
+      )
+      .finally(() => Deno.removeSync(file));
+  },
+});
+
+Deno.test({
+  name: "ASYNC: truncate file to a size of precisely len bytes",
+  async fn() {
+    const file: string = Deno.makeTempFileSync();
+    await Deno.writeFile(file, new TextEncoder().encode("hello world"));
+
+    await new Promise<void>((resolve, reject) => {
+      truncate(file, 3, (err: Error | null) => {
+        if (err !== null) reject();
+        else resolve();
+      });
+    })
+      .then(
+        () => {
+          const fileInfo: Deno.FileInfo = Deno.lstatSync(file);
+          assertEquals(fileInfo.size, 3);
+        },
+        () => {
+          fail("No error expected");
+        },
+      )
+      .finally(() => Deno.removeSync(file));
+  },
+});
+
+Deno.test({
+  name: "SYNC: truncate entire file contents",
+  fn() {
+    const file: string = Deno.makeTempFileSync();
+    truncateSync(file);
+    const fileInfo: Deno.FileInfo = Deno.lstatSync(file);
+    assertEquals(fileInfo.size, 0);
```

Comment:
@majidsajadi It seems that the errors from here will cause the leak of the temp file. How about surrounding these with `try { }` and calling `removeSync` in `finally {}`?